src/bitcoind.cpp

```diff
@@ -150,9 +159,20 @@ bool AppInit(int argc, char* argv[])
 
             pid_t sid = setsid();
             if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+                fprintf(stderr, "Error: setsid() failed: %s\n", strerror(errno));
+
+            (void)chdir("/");
```

Comment:
This may be a change of behavior: we should check whether there are arguments that take relative paths (e.g. `-datadir`, `-conf`, but also the `-*notify` options) that are affected by this.